frame/staking-rewards/src/lib.rs

```diff
@@ -370,21 +380,32 @@ pub mod pallet {
 		///  raises event of NFT `SplitCreation`
 		#[pallet::weight(10_000)]
 		pub fn split(
-			origin: OriginFor<T>,
-			asset: InstanceIdOf<T>,
-			amounts: BiBoundedVec<T::Balance, 2, 16>,
+			_origin: OriginFor<T>,
+			_asset: InstanceIdOf<T>,
+			_amounts: BiBoundedVec<T::Balance, 2, 16>,
 		) -> DispatchResult {
 			Err(DispatchError::Other("no implemented. TODO: call split on fnft provider"))
 		}
 
 		/// Extends fNFT position stake. Applied only to next epoch.
 		#[pallet::weight(10_000)]
+		#[transactional]
 		pub fn extend_stake(
 			origin: OriginFor<T>,
 			instance_id: InstanceIdOf<T>,
 			balance: T::Balance,
 		) -> DispatchResult {
-			Err(DispatchError::Other("no implemented. TODO: insert update for next fold"))
+			let owner = ensure_signed(origin)?;
+			let position = T::get_protocol_nft::<StakingNFTOf<T>>(&instance_id)?;
+			T::ensure_protocol_nft_owner::<StakingNFTOf<T>>(&owner, &instance_id)?;
+			let protocol_account = Self::account_id(&position.asset);
+			T::Assets::transfer(position.asset, &owner, &protocol_account, balance, false)?;
```

Comment:
Good to add a `keep_alive` parameter to the extrinsic :)

Reply:
ok.